src/structures/MessageButton.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+const BaseMessageComponent = require('./BaseMessageComponent');
+const { MessageButtonStyles, MessageComponentTypes } = require('../util/Constants.js');
+const Util = require('../util/Util');
+
+/**
+ * Represents a Button message component.
+ * @extends {BaseMessageComponent}
+ */
+class MessageButton extends BaseMessageComponent {
+  /**
+   * @typedef {BaseMessageComponentOptions} MessageButtonOptions
+   * @property {string} [label] The text to be displayed on this button
+   * @property {string} [customID] A unique string to be sent in the interaction when clicked
+   * @property {MessageButtonStyleResolvable} [style] The style of this button
+   * @property {Emoji} [emoji] The emoji to be displayed to the left of the text
+   * @property {string} [url] Optional URL for link-style buttons
+   * @property {boolean} [disabled=false] Disables the button to prevent interactions
+   */
+
+  /**
+   * @param {MessageButton|MessageButtonOptions} [data={}] MessageButton to clone or raw data
+   */
+  constructor(data = {}) {
+    super({ type: 'BUTTON' });
+
+    this.setup(data);
+  }
+
+  setup(data) {
+    /**
+     * The text to be displayed on this button
+     * @type {?string}
+     */
+    this.label = data.label ?? null;
+
+    /**
+     * A unique string to be sent in the interaction when clicked
+     * @type {?string}
+     */
+    this.customID = data.custom_id ?? data.customID ?? null;
+
+    /**
+     * The style of this button
+     * @type {MessageButtonStyle}
+     */
+    this.style = data.style ? MessageButton.resolveStyle(data.style) : null;
+
+    /**
+     * Emoji for this button
+     * @type {?Emoji|string}
+     */
+    this.emoji = data.emoji ?? null;
+
+    /**
+     * The URL this button links to, if it is a Link style button
+     * @type {?string}
+     */
+    this.url = data.url ?? null;
+
+    /**
+     * Whether this button is currently disabled
+     * @type {?boolean}
+     */
+    this.disabled = data.disabled ?? false;
+  }
+
+  /**
+   * Sets the custom ID of this button
+   * @param {string} customID A unique string to be sent in the interaction when clicked
+   * @returns {MessageButton}
+   */
+  setCustomID(customID) {
+    this.customID = Util.resolveString(customID);
+    return this;
+  }
+
+  /**
+   * Sets the interactive status of the button
+   * @param {boolean} disabled Whether this button should be disabled
+   * @returns {MessageButton}
+   */
+  setDisabled(disabled) {
+    this.disabled = disabled;
+    return this;
+  }
+
+  /**
+   * Set the emoji of this button
+   * @param {EmojiIdentifierResolvable} emoji The emoji to be displayed on this button
+   * @returns {MessageButton}
+   */
+  setEmoji(emoji) {
+    emoji = Util.parseEmoji(emoji);
```

Comment:
```suggestion
    this.emoji = Util.parseEmoji(emoji);
```
and line below removed too